src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj

```diff
@@ -3,8 +3,9 @@
   <Import Project="Sdk.props" Sdk="Microsoft.DotNet.SharedFramework.Sdk" />
 
   <PropertyGroup>
-    <SkipBuild Condition="'$(RuntimeFlavor)' == 'Mono'">true</SkipBuild>
-    <PlatformPackageType>RuntimePack</PlatformPackageType>
+    <!-- Crossgen is not used for Mono, and does not currently create freebsd packages -->
+    <SkipBuild Condition="'$(RuntimeFlavor)' == 'Mono' or ('$(DotNetBuildFromSource)' != 'true' and '$(RuntimeIdentifier)' == 'freebsd-x64')">true</SkipBuild>
```

Comment:
If CG2 is using live built of NativeAOT, or live build of singlefilehost, it will not need to restore `freebsd-x64` CG2 package right? If not, please keep the conditions on line 22.

nit: source build does not support anything other than linux, so `DotNetBuildFromSource` part is not necessary.

Reply:
CG2 support was enabled for freebsd here https://github.com/dotnet/runtime/commit/3c63559029276fea97633ea6115bcb9acb2cffe2. If we are using live build of host, we can enable package creating when target is FreeBSD as well.

For platforms which do not have CG2 support enabled (line 22), we can exclude them instead.

Reply:
I think if we avoid the apphost restore this could work, but I haven't tried it with your changes. My preference is to get this through, then cleanup and integrate with your change afterwards.

Reply:
Sure, other PR (#66814) can wait until this is completed. (I was thinking/hoping that we will get
non-NativeAOT-supported platforms build with singlefilehost after this so I will simply drop the second commit 🙂)